src/backend/probe/x86_64.rs

```diff
@@ -20,8 +26,8 @@ impl Vendor {
         let name = std::str::from_utf8(&name[..]).unwrap();
 
         Ok(match name {
-            "AuthenticAMD" => Self::Amd,
-            "GenuineIntel" => Self::Intel,
+            AUTHENTIC_AMD => Self::Amd,
+            GENUINE_INTEL => Self::Intel,
```

Comment:
I know it is old code, but could you please also fix the `get()` method into something like this, which does not have `transmute` and `from_utf8().unwrap`

```rust
    pub fn get() -> Result<Self> {
        let res = unsafe { __cpuid_count(0x00000000, 0x00000000) };

        let name = [
            res.ebx.to_le_bytes(),
            res.edx.to_le_bytes(),
            res.ecx.to_le_bytes(),
        ]
        .concat();

        if name.eq(AUTHENTIC_AMD.as_bytes()) {
            Ok(Self::Amd)
        } else if name.eq(GENUINE_INTEL.as_bytes()) {
            Ok(Self::Intel)
        } else {
            Err(Error::new(
                ErrorKind::Other,
                format!("unsupported vendor: {:#?}", name),
            ))
        }
    }
```